### DISCUSSION TITLE
Unable to pipe data on Windows

### DISCUSSION THREAD

**Discussion by vladdeSV** [jotaen/klog#379] [Bugs]
I am attempting to pipe output to `klog.exe` on Windows, version `v2.2  [0bb0d83]`.

In this example I receive the following error:
```sh
$ cat .\time.klg | .\klog.exe total
Error: No input given
Please do one of the following:
    a) pass one or multiple file names as argument
    b) pipe file contents via stdin
    c) specify a bookmark to read from by default
```

This was attempted using *PowerShell 7 (Preview)* and *Git for Windows (bash)* respectively.
<table>
<tr>
<td>
<img src="https://user-images.githubusercontent.com/7542961/116012603-be36d080-a62b-11eb-8495-c99f7062e89a.png">
</td>
<td>
<img src="https://user-images.githubusercontent.com/7542961/116012602-bbd47680-a62b-11eb-9e15-3df15627482d.png">
</td>
</tr>
</table>

**Comment by vladdeSV** [comment]
Just to be sure, in *Git for Windows (bash)* I ran the following command

```sh
$ cat ./time.klg | vim -
```

, which successfully piped *stdin* to *vim*.

![Screenshot 2021-04-26 011230](https://user-images.githubusercontent.com/7542961/116012702-83816800-a62c-11eb-8f21-d938dfadad38.png)

**Comment by vladdeSV** [comment]
Edit: Forgot to add *klog* version `v2.2  [0bb0d83]`. (i am sick and very tired)

**Comment by vladdeSV** [comment]
I also attempted this using pure `cmd.exe` (without using Windows Terminal) but received the same error

```cmd
C:\Users\vladde>echo "2021-04-26" | .\klog.exe total
Error: No input given
Please do one of the following:
    a) pass one or multiple file names as argument
    b) pipe file contents via stdin
    c) specify a bookmark to read from by default

C:\Users\vladde>
```

**Comment by jotaen** [comment]
Thanks for reporting, I can reproduce this.

**Comment by vladdeSV** [comment]
I still experience this issue, even after compiling newest version.

```sh
PS C:\Users\vladde> klog version
Command line tool: v2.2  [0bb0d83]
File format: version 1 (RFC)

Checking for newer version...
You already have the latest version!


PS C:\Users\vladde> echo 2020-01-01 | klog json
Error: No input given
Please do one of the following:
    a) pass one or multiple file names as argument
    b) pipe file contents via stdin
    c) specify a bookmark to read from by default
PS C:\Users\vladde>
```

Tested with *PowerShell*, *Git for Windows (bash)*, and pure *cmd.exe* on *Windows 10 Pro* (19042.928).

**Comment by vladdeSV** [comment]
If I can help with testing and so forth, tell me and I'll do my best to help out.

**Comment by jotaen** [comment]
Are you certain you have run the compiled binary and not your globally installed one? Because for the self-compiled binary the version output says `Command line tool: v?.?  [???????]` (unless you configure the build flags manually to set these values explicitly).

**Comment by vladdeSV** [comment]
You are completely right. I did not properly update my binary.

Works perfectly now 👍
